tuxemon/core/tools.py

```diff
@@ -385,7 +386,28 @@ def calc_dialog_rect(screen_rect):
     return rect
 
 
-def open_dialog(game, text, menu=None):
+def get_dialog_avatar(game, avatar):
+    """Gets the avatar sprite of a monster or NPC.
+
+    If avatar is a number, we're referring to a monster slot in the player's party
+    If avatar is a string, we're referring to a monster by name
+    TODO: If the monster name isn't found, we're referring to an NPC on the map
+
+    :rtype: Pygame surface
+    :returns: The surface of the monster or NPC avatar sprite
+    """
+    if avatar.isdigit():
+        player = game.player1
+        slot = int(avatar)
+        return player.monsters[slot].get_sprite("menu")
+    else:
+        avatar_monster = tuxemon.core.monster.Monster()
+        avatar_monster.load_from_db(avatar)
+        avatar_monster.flairs = {} # Don't use random flair graphics
+        return avatar_monster.get_sprite("menu")
+
+
+def open_dialog(game, text, avatar=None, menu=None):
```

Comment:
please search for "open_dialog" and double check that you've found and fixed all the times it is used.  because you have changed the order of the parameters, there may be bugs now.  python isn't static typed, so you won't find the bug until the game is running.  please resolve this after youve looked again.

Reply:
alternatively, you could go back to the original order and save yourself the trouble.

Reply:
Will take an extra close look after I figure out the animation issue. Already looked but will be extra sure.

Reply:
I did a grep through the whole code and looked carefully: All instances of open_dialog that specified the menu parameter were adjusted, the parameters should be in order everywhere. Even if somehow I managed to miss the obvious, I'm always around and could fix it at once.